working_villagers/jobs/hider.lua

```diff
@@ -0,0 +1,183 @@
+-- TODO it kinda works
+
+
+local func = working_villages.require("jobs/util")
+local hider_v1 = working_villages.require("jobs/hider_v1")
+
+-- TODO I'm sure there's a way to call get_nearest_player() instead
+local function get_nearest_player2(range_distance,pos)
+  local min_distance = range_distance
+  local player,ppos
+  local position = pos
+
+  local all_objects = minetest.get_objects_inside_radius(position, range_distance)
+  for _, object in pairs(all_objects) do
+    if object:is_player() then
+      local player_position = object:get_pos()
+      local distance = vector.distance(position, player_position)
+
+      if distance < min_distance then
+        min_distance = distance
+        player = object
+        ppos = player_position
+      end
+    end
+  end
+  return player,ppos,min_distance
+end
+
+local function find_stone_node(self)
+	local function foo(pos)
```

Comment:
You can write `return (function(self) ...)` if you want to skip creating a variable.

Reply:
thanks. I'm still really new to lua, so any tips are greatly appreciated.